brave/brave-core
# Add support for ephemeral P3A metrics, add ads enabled installation time metric

Resolves brave/brave-browser#28109

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start browser with fresh profile. Ensure metric does not exist in local state page.
1. Enable ads, ensure metric has value of 0.
2. Use MITM proxy to ensure metric is sent, ensure metric is marked as sent on local state page.
3. Close browser, move system time ahead by one calendar month, launch browser. Ensure metric is no longer present on local state page.
4. Create new profile, enable ads, ensure metric does not appear on local state page
5. Retest the above steps, but move the system time ahead by various amounts before step 1, in order to test different metric values.
